pkg/webhooks/pod/pod.go

```diff
@@ -0,0 +1,174 @@
+package pod
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"sync"
+
+	responsehelper "github.com/openshift/managed-cluster-validating-webhooks/pkg/helpers"
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+	"k8s.io/api/admission/v1beta1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName         string = "pod-validation"
+	privilegedNamespace string = `(^kube$|^kube-.*|^openshift$|^openshift-.*|^default$|^redhat-.*)`
+	exceptionNamespace  string = `(openshift-logging|openshift-operators)`
+)
+
+var (
+	privilegedNamespaceRe = regexp.MustCompile(privilegedNamespace)
+	exceptionNamespaceRe  = regexp.MustCompile(exceptionNamespace)
+	log                   = logf.Log.WithName(WebhookName)
+
+	scope = admissionregv1.NamespacedScope
+	rules = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{admissionregv1.OperationAll},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"v1"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"pods"},
+				Scope:       &scope,
+			},
+		},
+	}
+)
+
+type PodWebhook struct {
+	mu sync.Mutex
+	s  runtime.Scheme
+}
+
+// TimeoutSeconds implements Webhook interface
+func (s *PodWebhook) TimeoutSeconds() int32 { return 1 }
+
+// MatchPolicy implements Webhook interface
+func (s *PodWebhook) MatchPolicy() admissionregv1.MatchPolicyType {
+	return admissionregv1.Equivalent
+}
+
+// Name implements Webhook interface
+func (s *PodWebhook) Name() string { return WebhookName }
+
+// FailurePolicy implements Webhook interface
+func (s *PodWebhook) FailurePolicy() admissionregv1.FailurePolicyType {
+	return admissionregv1.Ignore
+}
```

Comment:
This should have a comment about why we are ignoring it, and how important it is that this doesn't change.